code/modules/admin/poll_management.dm

```diff
@@ -0,0 +1,735 @@
+	var/poll_id
+	///The type of poll to be created, must be POLLTYPE_OPTION, POLLTYPE_TEXT, POLLTYPE_RATING, POLLTYPE_MULTI or POLLTYPE_IRV.
+	var/poll_type
+	///Count of how many players have voted or responded to this poll.
+	var/poll_votes
+	///Ckey of the poll's original author
+	var/created_by
+	///Date and time the poll opens, timestamp format is YYYY-MM-DD HH:MM:SS.
+	var/start_datetime
+	///Date and time the poll will run until, timestamp format is YYYY-MM-DD HH:MM:SS.
+	var/end_datetime
+	///The title text of the poll, shows up on the list of polls.
+	var/question
+	///Supplementary text displayed only when responding to a poll.
+	var/subtitle
+	///Hides the poll from any client without a holder datum.
+	var/admin_only
+	///The number of responses allowed in a multiple-choice poll, more can be selected but won't be recorded.
+	var/options_allowed
+	///Hint for statbus, not used by the game; Stops the results of a poll from being displayed until the end_datetime is reached.
+	var/dont_show
+	///Allows a player to change their vote to a poll they've already voted on, off by default.
+	var/allow_revoting
+	///Indicates if a poll has been submitted or loaded from the DB so the management panel will open with edit functions.
+	var/edit_ready = FALSE
+	///Holds duration data when creating or editing a poll and refreshing the poll creation window.
+	var/duration
+	///Holds interval data when creating or editing a poll and refreshing the poll creation window.
+	var/interval
+	///Indicates a poll is set to not start in the future, still visible for editing but not voting on.
+	var/future_poll
+
+/**
+  * Datum which holds details of a poll option loaded from the database.
+  *
+  * Used to minimize the need for querying this data every time it's needed.
+  *
+  */
+/datum/poll_option
+	///Reference to the poll this option belongs to
+	var/datum/poll_question/parent_poll
+	///Table id of this option, will be null until poll has been created.
+	var/option_id
+	///Description/name of this option
+	var/text
+	///For rating polls, the minimum selectable value allowed; Supported value range is -2147483648 to 2147483647
+	var/min_val
+	///For rating polls, the maximum selectable value allowed; Supported value range is -2147483648 to 2147483647
+	var/max_val
+	///Optional for rating polls, description shown next to the minimum value
+	var/desc_min = ""
+	///Optional for rating polls, description shown next to the rounded whole middle value
+	var/desc_mid = ""
+	///Optional for rating polls, description shown next to the maximum value
+	var/desc_max = ""
+	///Hint for statbus, not used by the game; If this option should be included by default when calculating the resulting percentages of all options for this poll
+	var/default_percentage_calc
+
+/**
+  * Shows a list of all current and future polls and buttons to edit or delete them or create a new poll.
+  *
+  */
+/datum/admins/proc/poll_list_panel()
```

Comment:
Should have been tgui and it has common holder now but oh well, can be done later.

Reply:
Afaik we're still not doing tgui for admin interfaces for the sake of ensured compatability.